src/coreclr/tools/aot/ILLink.Shared/Annotations.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using ILLink.Shared.TrimAnalysis;
+
+// This is needed due to NativeAOT which doesn't enable nullable globally yet
+#nullable enable
+
+namespace ILLink.Shared
+{
+	// Temporary workaround - should be removed once linker can be upgraded to build against
+	// high enough version of the framework which has this enum value.
+	internal static class DynamicallyAccessedMemberTypesOverlay
+	{
+		public const DynamicallyAccessedMemberTypes Interfaces = (DynamicallyAccessedMemberTypes) 0x2000;
+	}
+
+	internal static class Annotations
+	{
+		public static bool SourceHasRequiredAnnotations (
+			DynamicallyAccessedMemberTypes sourceMemberTypes,
+			DynamicallyAccessedMemberTypes targetMemberTypes,
+			out string missingMemberTypesString)
+		{
+			missingMemberTypesString = string.Empty;
+
+			var missingMemberTypes = GetMissingMemberTypes (targetMemberTypes, sourceMemberTypes);
```

Comment:
The idea was to keep the files which come from linker repo **exactly** the same files - so that updating them is easy and all diffs are clearly visible. Also, updates to these files should ideally be done in the linker repo first (to keep one primary source).

So personally I would prefer we keep these files formatted as per dotnet/linker rules for now (and update the `.editorconfig`). I plan to move linker to runtime repo once we branch for .NET 8 and then we would reformat everything to match the runtime style (... can't wait ...).

But that's just my opinion, if you think it's more important to match the runtime style I'll change it.

Reply:
I think it's likely the odd code style choices from dotnet/linker repo are going to interfere with various repo-wide cleanups people sometimes do (using tabs instead of spaces is the main one that I can see this directory fighting with). Just raising this.